lib/foreman_maintain/utils/system_service.rb

```diff
@@ -0,0 +1,156 @@
+module ForemanMaintain::Utils
+  class SystemService
+    include Comparable
+
+    attr_reader :name, :priority, :component
+
+    def initialize(name, priority, options = {})
+      @sys = SystemHelpers.new
+      @name = name
+      @priority = priority
+      @component = options.fetch(:component, nil)
+    end
+
+    def <=>(other)
+      @priority <=> other.priority
+    end
+
+    def command(action, options = {})
+      do_wait = options.fetch(:wait, true) # wait for service to start
+      if do_wait && File.exist?('/usr/sbin/service-wait')
+        "service-wait #{@name} #{action}"
+      else
+        "systemctl #{action} #{@name}"
+      end
+    end
+
+    def status
+      execute('status')
+    end
+
+    def start
+      execute('start')
+    end
+
+    def stop
+      execute('stop')
+    end
+
+    def enable
+      execute('enable', :wait => false)
+    end
+
+    def disable
+      execute('disable', :wait => false)
+    end
+
+    def running?
+      status.first == 0
+    end
+
+    def to_s
+      @name + (@component ? " (#{@component})" : '')
+    end
+
+    def inspect
+      component = @component ? ':' + @component : ''
+      "SystemService(#{@name}#{component} [#{@priority}])"
+    end
+
+    def exist?
+      if @sys.systemd_installed?
+        systemd = @sys.execute("systemctl is-enabled #{@name} 2>&1 | tail -1").strip
+        systemd == 'enabled' || systemd == 'disabled'
+      else
+        File.exist?("/etc/init.d/#{service}")
+      end
+    end
+
+    def matches?(service)
+      if service.is_a? String
+        service == @name
+      else
+        service.name == @name && service.component == @component
+      end
+    end
+
+    private
+
+    def execute(action, options = {})
+      @sys.execute_with_status(command(action, options))
+    end
+  end
+
+  class RemoteDBService < SystemService
```

Comment:
The `RemoteDBService` name is misleading: when I was reading the code, my first impression was this represents only remote database (and was expecting `LocalDBService` counterpart).

What about `RemotableDBService`, or just `DBService`, and add some additoinal doc comment describing briefly the purpose.

Reply:
I was not sure about this concept either. I'm thinking of making it remote DB only and move the remote/local condition to the system_service factory.